### PR TITLE
Added a Spring Boot Sample integration

### PR DESCRIPTION
**Pull request by ricardozanini**
Hi! This is a PR to add a Spring Boot sample to the "samples" directory.

### CODE REVIEW

File: examples/features/standard/spring-boot-integration/pom.xml [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+			<artifactId>artemis-amqp-protocol</artifactId>
+			<version>${project.version}</version>
+		</dependency>
+		<dependency>
+			<groupId>org.amqphub.spring</groupId>
+			<artifactId>amqp-10-jms-spring-boot-starter</artifactId>
+			<version>${amqp-10-jms-spring.version}</version>
+		</dependency>
+		<dependency>
+			<groupId>org.messaginghub</groupId>
```
Comment (michaelandrepearce) [inline examples/features/standard/spring-boot-integration/pom.xml]: This does seem like If this is a spring boot example, and example using Spring JMS CachingConnection Factory would be more suitable. As spring users use this.

**Comment by ricardozanini** [comment]
HI @michaelandrepearce! Thanks for your review.

Regarding the sample and the differences between JMS Pool and Spring Caching feature, I'm going to edit the README file to enhance it with this details. I added the JMS Pool here because the [amqp-10-jms-spring-boot](https://github.com/amqphub/amqp-10-jms-spring-boot) itself is relying on it ([or if a reference to their classes are found in the classpath](https://github.com/amqphub/amqp-10-jms-spring-boot/blob/master/amqp-10-jms-spring-boot-autoconfigure/src/main/java/org/amqphub/spring/boot/jms/autoconfigure/AMQP10JMSAutoConfiguration.java#L58)). 

I may ask to the author why he's using JMS Pool instead of Spring's own feature. 

> I mostly have an issue here that Pooled CF you're bringing in from your side project (messaginghub) is originally Apache ActiveMQ 5 code, that rather than you contributing any fixes or enhancements to this and porting either to a subproject in ActiveMQ or to Artemis as a submodule, youve decided to host outside Apache, and lose all the attribution that others have made over time, https://github.com/apache/activemq/commits/master/activemq-jms-pool

I'm sorry, but I'm not the author of this project. If the code is came from ActiveMQ, I understand that you're right and the code should be in here. I think we might need to contact the author.

**Comment by jbertram** [comment]
IMO, the JMS pool from 5.x should not be migrated to Artemis.  It belongs in it's own project with it's own release cycle.  Also, it makes sense for it to *not* be in the ActiveMQ project to make clear that the pool is generic and isn't tied to any ActiveMQ broker.

The pool on messaginghub has JMS 2.0 support.

**Comment by michaelandrepearce** [comment]
@jbertram my issues are as follows:

The code broken out doesn't attribute the original contributors over time, people who contribute have a right to keep their attribution.

Peoples contributions were to the ASF, e.g. i contribute to activemq, on the basis my contribution remains in ASF, and i license my contribution to the foundation.

I get that it has JMS 2.0 support, but thats should be something that in ActiveMQ i think should be supported and i think really its a shame people didnt contribute back to Apache Software Foundation. If this is really needed we should look to have in Artemis (which is the JMS 2.0 broker) or we can have a submodule like we have CMS and NMS.

**Comment by jbertram** [comment]
I can't speak the contributor rights and licensing issues as I've not looked into the fine print on that, but I would say that the comparison to CMS and NMS isn't a good one since those are both tied directly to the 5.x broker whereas the pool isn't.  The pool is generic and could be used with any JMS client.

I'm not involved at all with the pool so I can't identify the exact, original motivations for breaking it into it's own project, but from an outsider's perspective it makes sense to me.  Clearly you disagree, but I'm not convinced that should stop this example from being merged.

**Comment by ricardozanini** [comment]
Justification behind use of JMS Pool:

> The JMS pool provides configuration capabilities not available in the CachingConnectionFactory and prevent certain not ideal uses like caching consumer which one should really never do. The code does not preclude you from using the CachingConnectionFactory at all and you can use standard Spring bean trickery to control what is created in you app if you want so if you really need to use CachingConnectionFactory. The PooledJMS library also provides JMS 2.0 functionality which is useful for those want that.

I'm going to add it to the README to make it clear to users. I'll correct the Travis issues and stash the commit to be consider to be merged.

Thanks @michaelandrepearce and @jbertram!

**Comment by michaelandrepearce** [comment]
The point with the example is essentially backing by adding the depenency that this is acceptable.

To me its not.

**Comment by ricardozanini** [comment]
I'm not in position to decide, but I could remove it and them add to the docs that one could use it for it's own use case.

**Comment by ricardozanini** [comment]
@jbertram I will remove the dependency from PooledJMS. But I can mention it in the docs? Or I remove it at all?

**Comment by jbertram** [comment]
@ricardozanini, I'd remove it completely from everything.

**Comment by ricardozanini** [comment]
@jbertram , @michaelandrepearce I've removed PooledJMS completely. Now users would rely only on amqp-starter for this sample. I couldn't found out what is the Travis issue. If someone point me out to the right direction, I could solve that as well. :)

**Comment by jbertram** [comment]
@ricardozanini, you should squash all of your commits so there's just one.

**Comment by ricardozanini** [comment]
Hi @jbertram! Done!

**Comment by michaelandrepearce** [comment]
@ricardozanini thanls for updating
 can you fix the rat test failures, essentailly missing license headers in files.

**Comment by ricardozanini** [comment]
@michaelandrepearce sure! But the only file without the header is the `additional-spring-configuration-metadata.json`, which I think won't allow comments. How do you guys add license headers in json files? Also, do we have some tooling for this type of validation before submitting a PR? For example, in Apache Camel I run checkstyle to guarantees that the code is ok.

**Comment by michaelandrepearce** [comment]
Just run mvn clean install :)

**Comment by ricardozanini** [comment]
Nice! But how do I add license header to json files? :(

**Comment by jdanekrh** [comment]
You cannot add comments to json files, due to https://plus.google.com/+DouglasCrockfordEsq/posts/RK8qyGVaGSr

It depends what does the parsing. Some parsers do recognize a comment syntax, such as the one used in vscode, as whitnessed by https://github.com/apache/activemq-artemis/blob/781e5e64a42f7cbbfac40294a6ce6e33074731ba/examples/protocols/amqp/dotnet/.vscode/tasks.json

**Comment by ricardozanini** [comment]
@jdanekrh thanks for your inputs! I've known about the json comments, but I was wondering if you guys had some workaround. I didn't know about the reference you pointed out. Nice!

@michaelandrepearce I've followed your tip and run `mvn clean install`. Went fine. :)
In Camel we have a specific profile for running those checks (besides RAT): `mvn clean install -Psourcecheck`

**Comment by michaelandrepearce** [comment]
Looks like possibly unrelated fail. Annoying. Could you update commit message or something and do a force push to try trigger another build.

**Comment by michaelandrepearce** [comment]
I built your branch locally fine, i think its just the pr build being its temperamental self. If you could just do the mentioned re push to trigger a rebuild just to get a green build.

**Comment by ricardozanini** [comment]
:dancer:

**Comment by michaelandrepearce** [comment]
Wahoo! Ill merge tonight. Thanks for the contribution and sticking with us!

**Comment by ricardozanini** [comment]
@michaelandrepearce sure!!! Count on me!

**Comment by michaelandrepearce** [comment]
@ricardozanini on merging it was noticed it was missing a JIRA and commit didnt prefix with the JIRA id. Next time if you could do this.

This time round to merge and not cause an issue, I created a JIRA for this https://issues.apache.org/jira/projects/ARTEMIS/issues/ARTEMIS-2033, and on merging i updated your commit message to be prefixed with the JIRA id.

**Comment by ricardozanini** [comment]
@michaelandrepearce Ok! Yes, after sending the PR I've noticed that. Sorry, next time I'll do it the right way.

**Pull request closed without merging**